src/config/interface.ts

```diff
@@ -0,0 +1,53 @@
+type custom = {
+  name: string;
+  given_name: string;
+  family_name: string;
+  nickname: string;
+  email: string;
+  email_verified: string;
+  picture: string;
+};
+
+export interface decoded_jwt {
+  aud: string[];
+  email: string;
+  exp: number;
+  iat: number;
+  nbf: number;
+  iss: string;
+  type: string;
+  identity_nonce: string;
+  custom?: custom; //custom object will only exist for authentication with Auth0
+  sub: string;
+  country: string;
+}
+
+export interface auth_status {
+  jwt: string;
+  decoded_jwt?: decoded_jwt;
+  idp?: string; // Auth0 or GSuite
```

Comment:
```suggestion
  idp?: "auth0" | "gsuite";
```

Reply:
Def good suggestion but this logic is going to change very quickly in our phase 2 which is to implement an admin page. It turns out that CloudFlare Access supports tokens signed to _more_ than one `idp` at a time. The current implementation has an admin page (ish) but because it swaps out the `idp` from auth0 to gsuite so the person logged in loses access to the regular pages & when our code prompts cloudflare to reauthenticate when the person tries to access a regular page it won't happen since the person is technically signed into both from their perspective.

The very very near future fix coming to this is to change it to `idp?: string[]` so that we can store info for officers that might have one token for 2 idps with 2 audiences listed.